[('amount', 500), ('product', <Product: Product object (3)>)]
[('amount', 500), ('product', <Product: Product object (5)>)]
[('amount', 500), ('product', <Product: Product object (3)>)]
[('amount', 500), ('product', <Product: Product object (5)>)]
[('amount', 2), ('product', <Product: Product object (4)>)]
[('amount', 499), ('product', <Product: Product object (4)>)]
Not Found: /api/wharehouse/
Not Found: /favicon.ico
Internal Server Error: /api/warehouse
Traceback (most recent call last):
  File "C:\Users\edgar\Documents\python\trydjango\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\edgar\Documents\python\trydjango\Lib\site-packages\django\utils\deprecation.py", line 135, in __call__
    response = self.process_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\edgar\Documents\python\trydjango\Lib\site-packages\django\middleware\common.py", line 54, in process_request
    path = self.get_full_path_with_slash(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\edgar\Documents\python\trydjango\Lib\site-packages\django\middleware\common.py", line 88, in get_full_path_with_slash
    raise RuntimeError(
RuntimeError: You called this URL via POST, but the URL doesn't end in a slash and you have APPEND_SLASH set. Django can't redirect to the slash URL while maintaining POST data. Change your form to point to 127.0.0.1:8000/api/warehouse/ (note the trailing slash), or set APPEND_SLASH=False in your Django settings.
